api-test/hasura/actions/_handlers/createEpoch.test.ts

```diff
@@ -0,0 +1,557 @@
+import { assert } from 'console';
+
+import { DateTime, Interval } from 'luxon';
+
+import { adminClient } from '../../../../api-lib/gql/adminClient';
+import { zEpochInputParams } from '../../../../api/hasura/actions/_handlers/createEpoch';
+import {
+  createCircle,
+  createProfile,
+  mockUserClient,
+  createUser,
+} from '../../../helpers';
+import { getUniqueAddress } from '../../../helpers/getUniqueAddress';
+
+let address, profile, circle, client;
+
+beforeEach(async () => {
+  address = await getUniqueAddress();
+  circle = await createCircle(adminClient);
+  profile = await createProfile(adminClient, { address });
+  await createUser(adminClient, { address, circle_id: circle.id });
+  client = mockUserClient({ profileId: profile.id, address });
+});
+
+describe('createEpoch', () => {
+  describe('invalid input', () => {
+    test('errors when another repeating epoch exists', async () => {
+      expect.assertions(1);
+      const now = DateTime.now().plus({ weeks: 2 });
+      const first = async () =>
+        client.mutate({
+          createEpoch: [
+            {
+              payload: {
+                circle_id: circle.id,
+                params: {
+                  type: 'custom',
+                  start_date: now.toISO(),
+                  end_date: now.plus({ days: 3 }).toISO(),
+                  frequency: 1,
+                  frequency_unit: 'weeks',
+                },
+              },
+            },
+            { __typename: true },
+          ],
+        });
+
+      const second = async () =>
+        client.mutate({
+          createEpoch: [
+            {
+              payload: {
+                circle_id: circle.id,
+                params: {
+                  type: 'custom',
+                  start_date: now.plus({ weeks: 4 }).toISO(),
+                  end_date: now.plus({ weeks: 4, days: 3 }).toISO(),
+                  frequency: 1,
+                  frequency_unit: 'weeks',
+                },
+              },
+            },
+            { __typename: true },
+          ],
+        });
+
+      await first();
+      return second().catch((e: any) => {
+        expect(e.response.errors[0].message).toContain(
```

Comment:
i wish GraphQLError's normal error message were also set so this could just be
```js
await expect(second).rejects.toThrow('You cannot have more than one repeating active epoch.')
```

Reply:
fwiw, i recently made the above change